lib/timeline/component/item/PointItem.js

```diff
@@ -104,40 +164,108 @@ PointItem.prototype.redraw = function() {
-    dom.dot.className  = 'vis-item vis-dot' + className;
-
-    // recalculate size of dot and contents
-    this.props.dot.width = dom.dot.offsetWidth;
-    this.props.dot.height = dom.dot.offsetHeight;
-    this.props.content.height = dom.content.offsetHeight;
-
-    // resize contents
-    if (this.options.rtl) {
-      dom.content.style.marginRight = 2 * this.props.dot.width + 'px';
-    } else {
-      dom.content.style.marginLeft = 2 * this.props.dot.width + 'px';
-    }
-    //dom.content.style.marginRight = ... + 'px'; // TODO: margin right
-
-    // recalculate size
-    this.width = dom.point.offsetWidth;
-    this.height = dom.point.offsetHeight;
-
-    // reposition the dot
-    dom.dot.style.top = ((this.height - this.props.dot.height) / 2) + 'px';
-    if (this.options.rtl) {
-      dom.dot.style.right = (this.props.dot.width / 2) + 'px';
-    } else {
-      dom.dot.style.left = (this.props.dot.width / 2) + 'px';
+    this.dom.point.className  = 'vis-item vis-point' + className;
+    this.dom.dot.className  = 'vis-item vis-dot' + className;
+  }
+}
+
+PointItem.prototype._getDomComponentsSizes = function() {
+  return {
+    dot:  {
+      width: this.dom.dot.offsetWidth,
+      height: this.dom.dot.offsetHeight
+    },
+    content: {
+      width: this.dom.content.offsetWidth,
+      height: this.dom.content.offsetHeight
+    },
+    point: {
+      width: this.dom.point.offsetWidth,
+      height: this.dom.point.offsetHeight
     }
+  }
+}
+
+PointItem.prototype._updateDomComponentsSizes = function(sizes) {
+  // recalculate size of dot and contents
+  this.props.dot.width = sizes.dot.width;
+  this.props.dot.height = sizes.dot.height;
+  this.props.content.height = sizes.content.height;
 
-    this.dirty = false;
+  // resize contents
+  if (this.options.rtl) {
+    this.dom.content.style.marginRight = 2 * this.props.dot.width + 'px';
+  } else {
+    this.dom.content.style.marginLeft = 2 * this.props.dot.width + 'px';
   }
-  
-  this._repaintOnItemUpdateTimeTooltip(dom.point);
+  //this.dom.content.style.marginRight = ... + 'px'; // TODO: margin right
+
+  // recalculate size
+  this.width = sizes.point.width;
+  this.height = sizes.point.height;
+
+  // reposition the dot
+  this.dom.dot.style.top = ((this.height - this.props.dot.height) / 2) + 'px';
+  if (this.options.rtl) {
+    this.dom.dot.style.right = (this.props.dot.width / 2) + 'px';
+  } else {
+    this.dom.dot.style.left = (this.props.dot.width / 2) + 'px';
+  }
+
+  this.dirty = false;
+}
+
+PointItem.prototype._repaintDomAdditionals = function() {
+  this._repaintOnItemUpdateTimeTooltip(this.dom.point);
   this._repaintDragCenter();
-  this._repaintDeleteButton(dom.point);
+  this._repaintDeleteButton(this.dom.point);
+}
+
+/**
+ * Repaint the item
+ * @param {boolean} [returnQueue=false]  return the queue
+ * @return {boolean} the redraw queue if returnQueue=true
+ */
+PointItem.prototype.redraw = function(returnQueue) {
```

Comment:
Strike three: identical to `BoxItem` and `BackgroundItem`. It's not a question any more: DRY this.

Reply:
Will be done in future PR